lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp

```diff
@@ -1752,17 +1752,16 @@ ConstString TypeSystemSwiftTypeRef::GetSwiftModuleFor(const SymbolContext &sc) {
   return {};
 }
 
-SwiftASTContext *TypeSystemSwiftTypeRef::GetSwiftASTContextFromExecutionScope(
+SymbolContext TypeSystemSwiftTypeRef::GetSymbolContext(
     ExecutionContextScope *exe_scope) const {
-  if (exe_scope) {
-    ExecutionContext exe_ctx;
-    exe_scope->CalculateExecutionContext(exe_ctx);
-    return GetSwiftASTContextFromExecutionContext(&exe_ctx);
-  }
-  return GetSwiftASTContextFromExecutionContext(nullptr);
+  if (!exe_scope)
+    return {};
+  ExecutionContext exe_ctx;
+  exe_scope->CalculateExecutionContext(exe_ctx);
+  return GetSymbolContext(&exe_ctx);
 }
 
-SwiftASTContext *TypeSystemSwiftTypeRef::GetSwiftASTContextFromExecutionContext(
+SymbolContext TypeSystemSwiftTypeRef::GetSymbolContext(
```

Comment:
These two `TypeSystemSwiftTypeRef::GetSymbolContext` don't seem to have much to do with `TypeSystemSwiftTypeRef` (they also don't access any instance state and could be static), I wonder if these could live in somewhere more else so they could be more discoverable and reusable (for example, as static functions in `SymbolContext`)?

Reply:
If we decide to do this these functions should probably de upstream.